### PR TITLE
Login admin interface

### PR DESCRIPTION
**Pull request by ulfgebhardt**
## 🍰 Pullrequest

Implements the login/authenticate mechanism for the new admin interface. Furthermore the wayback is implemented and a convenience mode to disabled auth for development

### Issues

- includes #1116 
- includes #1045 

### Todo

- [x] Convenience mode
- [x] `isAdmin` from backend
- [x] logout
- [x] back2wallet
- [x] verify login (requires #1095 )
- [x] frontend tests

### CODE REVIEW

File: frontend/src/components/SidebarPlugin/SideBar.vue [hunk 1]
```diff
@@ -54,6 +55,14 @@
           </li>
         </ul>
 
+        <ul class="navbar-nav ml-3" v-if="this.$store.state.isAdmin">
```
Comment (Mogge) [inline frontend/src/components/SidebarPlugin/SideBar.vue:58]: Please Test this

**Review by Mogge — CHANGES_REQUESTED** [review]

File: frontend/src/routes/guards.js [hunk 2]
```diff
@@ -1,12 +1,27 @@
...
     const publisherId = to.query.pid
     if (publisherId) {
       store.commit('publisherId', publisherId)
       delete to.query.pid
     }
-    // handle authentication
+    next()
+  })
+
+  // store token on authenticate
```
[the remark this answers is not in the document]
Comment (ulfgebhardt) [reply frontend/src/routes/guards.js:12]: It will bring you to `not-found` page

**Comment by Mogge** [comment]
I fixed the test and also checked that logout is dispatched. But I get the following, when clicking on admin interface
```
message | "Controller class Admin could not be found."
-- | --
url | "/admin/authenticate?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwdWJLZXkiOiJlMDlhNzk2NTQ1NTM3NTI2YTZkMTFhZjc0OTg1NDFmZjg2MGJkZTg0MzExZTNkYTU4ZjI5Y2NiZWY4MjUzNzQ5IiwiaWF0IjoxNjM3NjY2NjkyLCJleHAiOjE2Mzc2NjcyOTJ9.mfvCZIcd_ub5BSxprLdC3iPUMXO_pWiKfIYlAipfdic"
code | 404
file | "/var/www/cakephp/vendor/cakephp/cakephp/src/Http/ControllerFactory.php"
line | 105
```
Looks like nginx is not working as expected. On my setup I can access localhost/admin

File: frontend/src/components/SidebarPlugin/SideBar.vue (hunk 1 above)
Comment (ulfgebhardt) [reply frontend/src/components/SidebarPlugin/SideBar.vue]: 🤷🏻

**Review by Mogge — APPROVED** [review]

**Pull request merged**